gcloud/_helpers.py

```diff
@@ -159,6 +163,34 @@ def _app_engine_id():
     return app_identity.get_application_id()
 
 
+def _file_project_id():
+    """Gets the project id from the credentials file if one is available.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from JSON credentials file if value exists,
+              else ``None``.
+    """
+    credentials_file_path = os.getenv(CREDENTIALS)
+    if credentials_file_path:
+        with open(credentials_file_path, 'rb') as credentials_file:
+            credentials = json.load(credentials_file)
+            return credentials.get('project_id')
+
+
+def _default_service_project_id():
+    """Retrieves the project ID from the gcloud command line tool.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from default configuration file else ``None``
+    """
+    full_config_path = os.path.expanduser(DEFAULT_CONFIGURATION_PATH)
+    config = configparser.RawConfigParser()
+    config.read(full_config_path)
```

Comment:
I just checked out the documentation for `config.read` and it's probably worth mentioning with a code comment.

```
In [15]: config.read?
Signature: config.read(filenames)
Docstring:
Read and parse a filename or a list of filenames.

Files that cannot be opened are silently ignored; this is
designed so that you can specify a list of potential
configuration file locations (e.g. current directory, user's
home directory, systemwide directory), and all existing
configuration files in the list will be read.  A single
filename may also be given.

Return list of successfully read files.
File:      /usr/lib/python2.7/ConfigParser.py
Type:      instancemethod
```

Sorry I didn't mention before. I checked the docs because I was worried an `OSError` for a non-existent file would cause an `OSError` instead of just moving on to the next check